sklearn/linear_model/logistic.py

```diff
@@ -1059,20 +1079,29 @@ def fit(self, X, y):
             classes_ = [None]
 
         for ind, class_ in enumerate(classes_):
+            if class_ is None or warm_start_coef is None:
+                # multinomial hack
+                warm_start_coef_ = warm_start_coef
+            elif warm_start_coef is not None:
+                warm_start_coef_ = warm_start_coef[ind]
+
             coef_, _ = logistic_regression_path(
                 X, y, pos_class=class_, Cs=[self.C],
                 fit_intercept=self.fit_intercept, tol=self.tol,
                 verbose=self.verbose, solver=self.solver,
                 multi_class=self.multi_class, max_iter=self.max_iter,
-                class_weight=self.class_weight)
+                class_weight=self.class_weight, random_state=self.random_state,
+                coef=warm_start_coef_)
             self.coef_.append(coef_[0])
 
         self.coef_ = np.squeeze(self.coef_)
-        # For the binary case, this get squeezed to a 1-D array.
+        # For the binary case, this get squeezed to a 1-D array,
+        # or even 0-D if n_features == 1
```

Comment:
`logistic_regression_path` returns a list of array, and `LogisticRegression` uses a list to append the lists of array obtained for the different classes.
In order to transform the list of array into an array, we use `np.squeeze()`.

However, if we have only 1 class, it get squeezed to much, so we need to manually add another dimension. If we have only 1 class and 1 feature, it get squeezed to a scalar, so we need to add 2 other dimensions.
Actually, I wonder if we should not also add the case with 1 feature and multiclass.